### PR TITLE
Add r2m2d2cp prior

### PR DESCRIPTION
**Pull request by ferrine**
I think I'll need to add tests and some examples

### CODE REVIEW

File: pymc_experimental/distributions/multivatiate.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        variance explained, should sum up to one, by default None
+    r2_std : tensor, optional
+        Optional uncertainty over :math:`R^2`, by default None
+    positive_probs : tensor, optional
+        Optional probability of variables contribution to be positive, by default 0.5
+    positive_probs_std : tensor, optional
+        Optional uncertainty over effect direction probability, by default None
+    centered : bool, optional
+        Centered or Non-Centered parametrization of the distribution, by default Non-Centered. Advised to check both
+
```
Comment (ricardoV94) [inline pymc_experimental/distributions/multivatiate.py:108]: Maybe add a code black with an example use?

File: pymc_experimental/distributions/__init__.py [hunk 2]
```diff
@@ -18,7 +18,9 @@
 """
 
 from pymc_experimental.distributions.continuous import GenExtreme
+from pymc_experimental.distributions.multivatiate import R2D2M2CP
 
 __all__ = [
     "GenExtreme",
+    "R2D2M2CP",
```
Comment (ricardoV94) [inline pymc_experimental/distributions/__init__.py:25]: Don't forget to add it to the docs

File: pymc_experimental/distributions/multivatiate.py (hunk 1 above)
Comment (ferrine) [reply pymc_experimental/distributions/multivatiate.py:108]: Yep, do we have pydoctest setup?

Comment (ricardoV94) [reply pymc_experimental/distributions/multivatiate.py:108]: I don't know, feel free to add it (and fix the existing docstrings haha)

File: pymc_experimental/distributions/__init__.py [hunk 3]
```diff
@@ -18,7 +18,11 @@
 """
 
 from pymc_experimental.distributions.continuous import GenExtreme
+from pymc_experimental.distributions.histogram_utils import histogram_approximation
+from pymc_experimental.distributions.multivatiate import R2D2M2CP
```
Comment (ColCarroll) [inline pymc_experimental/distributions/__init__.py:22]: typo on filename: `multivariate`

File: pymc_experimental/distributions/multivatiate.py [hunk 4]
```diff
@@ -0,0 +1,246 @@
+#   Copyright 2022 The PyMC Developers
```
Comment (twiecki) [inline pymc_experimental/distributions/multivatiate.py:1]: also in filename typo

File: pymc_experimental/distributions/__init__.py (hunk 2 above)
Comment (ferrine) [reply pymc_experimental/distributions/__init__.py:26]: I also added histogram_approximation to this scope as it was like this in docs

File: pymc_experimental/distributions/multivariate.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    --------
+    Here are arguments explained in a synthetic example
+
+    >>> import pymc_experimental as pmx
+    >>> import pymc as pm
+    >>> import numpy as np
+    >>> X = np.random.randn(10, 3)
+    >>> b = np.random.randn(3)
+    >>> y = X @ b + np.random.randn(10) * 0.04 + 5
+    >>> with pm.Model(coords=dict(variables=["a", "b", "c"])) as model:
```
Comment (ricardoV94) [inline pymc_experimental/distributions/multivariate.py:136]: Maybe split the initialization from the first model, so users can easily run the second or third examples if they want?

Also I think we have a different docs formatting that allows to copy paste snippets directly

Comment (ricardoV94) [reply pymc_experimental/distributions/multivariate.py:136]: https://github.com/pymc-devs/pymc-experimental/blob/e38da068d92e8464dc879790ecad69ebdf6d4398/pymc_experimental/marginal_model.py#L56

File: docs/api_reference.rst [hunk 6]
```diff
@@ -32,6 +32,18 @@ Distributions
...
+
+
+Gaussian Processess
+===================
+
+.. currentmodule:: pymc_experimental.gp
+.. autosummary::
+   :toctree: generated/
+
+   latent_approx.HSGP
```
Comment (ricardoV94) [inline docs/api_reference.rst:46]: Wasn't this removed?

**Comment by ricardoV94** [comment]
Looks good, just a question about one entry you added to the docs.

We should follow up with a PyMC examples showing this in practice, WDYT?

**Comment by ferrine** [comment]
Lets create an issue and I'll make that happen

File: docs/api_reference.rst (hunk 6 above)
Comment (ferrine) [reply docs/api_reference.rst:46]: I think it was

**Comment by ferrine** [comment]
I've reverted my changes in documentation to quite minimal, there are still issues left, but I guess they should be addressed in separate PRs

**Comment by ferrine** [comment]
This is ready to review

**Review by ricardoV94 — DISMISSED** [review]

**Comment by twiecki** [comment]
🔥

**Pull request merged**